examples/viewer.js

```diff
@@ -36,16 +38,239 @@ let droppedFramePerArray = [];
+            tooltip: 'Time duration the viewer was waiting for the master to start (time to start the SDK after the viewer signaling channel was connected)',
+            color: 'yellow',
+        }, 
+        signaling: {
+            name: 'signaling-viewer',
+            startTime: '',
+            endTime: '',
+            tooltip: 'Time taken to establish a signaling connection on the viewer-side',
+            color: '#F44336',
+        },
+        setupMediaPlayer: {
+            name: 'setup-media-player-viewer',
+            startTime: '',
+            endTime: '',
+            tooltip: 'Time taken to setup a media player on the viewer-side by seeking permissions for mic / camera (if needed), fetch tracks from the same and add them to the peer connection',
+            color: '#9575CD',
+        }, 
+        offAnswerTime: {
+            name: 'sdp-exchange-viewer',
+            startTime: '',
+            endTime: '',
+            tooltip: 'Time taken to send an offer and receive a response',
+            color: '#FF6F00',
+        },
+        describeChannel: {
+            name: 'signaling-viewer-describe-channel',
+            startTime: '',
+            endTime: '',
+            tooltip: 'Time taken for the API call to desribeSignalingChannel on the viewer',
+            color: '#EF9A9A',
+        },
+        describeMediaStorageConfiguration: {
+            name: 'signaling-viewer-describe-media-storage-config',
+            startTime: '',
+            endTime: '',
+            tooltip: 'Time taken for the API call to desribeSignalingChannel on the viewer',
+            color: '#EF9A9A',
+        }, 
+        channelEndpoint: {
+            name: 'signaling-viewer-get-signaling-channel-endpoint',
+            startTime: '',
+            endTime: '',
+            tooltip: 'Time taken for the API call to getSignalingChannelEndpoint on the viewer',
+            color: '#EF9A9A',
+        },
+        iceServerConfig: {
+            name: 'signaling-viewer-get-ice-server-config',
+            startTime: '',
+            endTime: '',
+            tooltip: 'Time taken for the API call to getIceServerConfig on the viewer',
+            color: '#EF9A9A',
+        },
+        connectAsViewer: {
+            name: 'signaling-connect-as-viewer',
+            startTime: '',
+            endTime: '',
+            tooltip: 'Time taken to open the websocket via connectAsViewer',
+            color: '#EF9A9A',
+        },
+        iceGathering: {
+            name: 'ice-gathering-viewer',
+            startTime: '',
+            endTime: '',
+            tooltip: 'Time taken to gather all ice candidates on the viewer',
+            color: '#90CAF9',
+        }, 
+        peerConnection: {
+            name: 'pc-establishment-viewer',
+            startTime: '',
+            endTime: '',
+            tooltip: 'Time taken to establish the peer connection on the viewer',
+            color: '#2196F3',
+        },
+        ttffAfterPc: {
```

Comment:
Since this is a metric we calculate in our samples on the C side, what would happen if customers do not include this? How would it show up in the graph?

Reply:
The customers cannot control excluding it, however, if the values are somehow altered in such a way that they cannot be displayed / are invalid, I have checks to skip them in the function at the end

Reply:
I mean, ttff is something that is added in our C samples (it is not within the SDK core). Customers have often excluded the calculation for this when they write their own sample. These are still metrics coming from C SDK right?

Reply:
ttff is not coming from the C SDK, it is viewer specific. It is the first-frame - viewerrButtonPressed